echopype/convert/set_groups_ek80.py

```diff
@@ -1311,31 +1318,91 @@ def set_vendor(self) -> xr.Dataset:
             ds_cal = ds_cal.rename_vars({"impedance": "impedance_transducer"})
 
         #  Save decimation factors and filter coefficients
-        coeffs = dict()
-        decimation_factors = dict()
+        coeffs_and_decimation = {
+            t: {FILTER_IMAG: [], FILTER_REAL: [], DECIMATION: []}
+            for t in [WIDE_BAND_TRANS, PULSE_COMPRESS]
+        }
+
         for ch in self.sorted_channel["all"]:
             # filter coeffs and decimation factor for wide band transceiver (WBT)
             if self.parser_obj.fil_coeffs and (ch in self.parser_obj.fil_coeffs.keys()):
```

Comment:
@leewujung do you know why this one checks for only `fil_coeffs` for WBT, even though you also appear to need the decimation value? Same thing below with PC, you only check for `fil_df` even though you also need `fil_coeffs`?

Reply:
I think for best practice both should be checked. These parameters are usually all or none, so if one does not exist it means the other also does not exist.

Reply:
> These parameters are usually all or none, so if one does not exist it means the other also does not exist.

To clarify, is this between the WBT and PC or between filter coefficient and decimation? WBT can't exist without PC? or filter coefficient can't exist without decimation?

Reply:
I think it is safer to check them separately:
- WBT filter coeff (and if it does not exist, WBT decimation factor likely won't exist)
- PC filter coeff (and if it does not exist, PC decimation factor likely won't exist)